File: fieldsync-inventory-backend/src/main/java/com/example/fieldsync_inventory_api/dto/stock/transaction_operation/stock_in/TransactionOperationStockInResDTO.java
package com.example.fieldsync_inventory_api.dto.stock.transaction_operation.stock_in;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import com.example.fieldsync_inventory_api.enums.OperationTask;
import com.example.fieldsync_inventory_api.enums.OperationType;

import java.time.Instant;
import java.util.ArrayList;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class TransactionOperationStockInResDTO {
    private Long transactionId;
    private OperationType type; // transaction types
    private OperationTask task; // movement types which are the same each
    private String supplier; // Purchase's supplier
    ArrayList<TOStockMovementStockInResDTO> stockMovements;
    private Instant transactionDate;
    private String description;
    private String performedBy; // user
}

File: fieldsync-inventory-app/app/src/main/java/com/example/fieldsync_inventory_app/data/remote/dto/transaction_operation/response/stock_in/TransactionOperationStockInResDTO.kt
package com.example.fieldsync_inventory_app.data.remote.dto.transaction_operation.response.stock_in

data class TransactionOperationStockInResDTO(
    val transactionId: Long,
    val type: String,// transaction types
    val task: String, // movement types which are the same value each
    val supplier: String, // Purchase's supplier
    val stockMovements: List<TOStockMovementStockInResDTO>,
    val transactionDate: String, // ISO 8601 string
    val description: String,
    val performedBy: String
)
